src/responder/kcm/kcmsrv_cmd.c

```diff
@@ -418,6 +430,19 @@ static errno_t kcm_recv_data(int fd, struct kcm_reqbuf *reqbuf)
         return ret;
     }
 
+    msglen = kcm_input_get_payload_len(reqbuf->v_len.kiov_base,
+                                       reqbuf->v_len.kiov_len);
```

Comment:
I think here we should still check if the msglen is within the acceptable limits so that some evil client doesn't try to send a message with 1TB size which might OOM the deamon.

Reply:
Fixed locally, thanks!